### ISSUE TITLE
do I need to modify this to use it with jenssegers/laravel-mongodb ?

### ISSUE THREAD

**Issue by vesper8** [GeneaLabs/laravel-model-caching#57]
I am using jenssegers/laravel-mongodb and using hybrid relations between mysql models and mongodb models

the mongodb models however rather than extend laravel's default model, extend a special wrapper model of sorts

a lot of model-related packages on github do work out of the box with mongodb models but I'm not sure how to use this one since it depends on extending a different model

any chance you could have a quick look at jenssegers/laravel-mongodb and see if there's a quick and easy solution to using it with your package?

thanks much!

**Comment by vesper8** [comment]
I've gone ahead and made a duplicate of your CachedModel and called ot MongoCachedModel,

Then inside that class I replaced the extend Model by extend Eloquent

And added use Jenssegers\Mongodb\Eloquent\Model as Eloquent;

That's all that was needed to get it to work with mongodb models

Except that now I can't use the artisan flush command on that model because it's not an instance of "CachedModel"

Can you think of a more elegant way to handle this?

**Comment by vesper8** [comment]
I also ran into one problem so far when working with mongodb models. Unlike with mysql, sometimes the values are arrays themselves and that triggered a Array to String conversion error on these lines inside the CacheKey

```
        $value = array_get($where, 'value');
        $value .= $this->getTypeClause($where);
        $value .= $this->getValuesClause($where);
```

I was able to fix it by changing it to

```
        $value = array_get($where, 'value');
        if (is_array($value)) {
            $value = implode("_", $value);
        }
        $value .= $this->getTypeClause($where);
        $value .= $this->getValuesClause($where);
```

**Comment by mikebronner** [comment]
Interesting, I will see about adding this. If you can submit a PR that would expedite things, but it sounds like a good idea.

**Comment by vesper8** [comment]
hey @mikebronner I'm hoping maybe you can think of something here...

what I mentioned above is really the only small change I needed to do in order to get it to work with my mongodb models. This was working perfectly on my DEV environment. Mongo models were being invalidated exactly as you'd expect a mysql model to.. no biggie

but then tonight I deployed my changes to production.. and I'm baffled.

Mysql models are being invalidated, but not the mongo ones! And there are no errors being thrown at all.

I verified that I'm loading my fork correctly with that small change you see above. I cleared all caches.. opcache.. everything.. multiple times. I added \Log::debug in my _CachedMongoModel file which is a clone of CachedModel as I mentioned above. I added logging on the update/saved events and can confirm that it's going in there.. and it's executing the $instance->flushCache();.. and it's not throwing any errors but it won't flush the darn mongo models

I even connected through Medis.. a pretty cool Redis GUI client, and I compared the tag names on both my DEV and production.. and the tag names are the same. If I use the command line to flush models.. it works on DEV for both mysql and mongo tags.. but on Live it only works on mysql models

I made myself my own flush command btw because yours requires that the model be an instance of CachedModel so I just commented those out.

Anyway I am really out of ideas of why it isn't able to flush the mongo models on live.. the tags look identical.. ??

I had to revert the change now.. if you have any idea you'd really be helping me out here!

Thanks!

**Comment by mikebronner** [comment]
Fixed in version 0.2.30. Please re-open if still a problem.
